clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -71,12 +71,12 @@
     /**
      * @see KafkaConsumer#commit(Map, CommitType)
      */
-    public void commit(Map<TopicPartition, Long> offsets, CommitType commitType);
+    public void commit(Map<TopicPartition, OffsetMetadata> offsets, CommitType commitType);
```

Comment:
What about
    public void commit(List<OffsetMetadata> offsets, CommitType type)
where now the `OffsetMetadata` class includes the `TopicPartition`?